environment/src/main/java/com/sequenceiq/environment/environment/validation/network/AzureEnvironmentNetworkValidator.java

```diff
@@ -91,8 +88,43 @@ private void checkExistingNetworkParamsProvidedWhenSubnetsPresent(NetworkDto net
         }
     }
 
+    private void checkNetworkIdIsSpecifiedWhenSubnetIdsArePresent(ValidationResultBuilder resultBuilder,
+            AzureParams azureParams, NetworkDto networkDto) {
+        if (StringUtils.isEmpty(azureParams.getNetworkId()) && CollectionUtils.isNotEmpty(networkDto.getSubnetIds())) {
+            resultBuilder.error("If subnetIds are specified, then networkId must be specified too.");
+        }
+    }
+
+    private void checkSubnetsProvidedWhenExistingNetwork(ValidationResultBuilder resultBuilder,
+            AzureParams azureParams, Map<String, CloudSubnet> subnetMetas) {
+        if (StringUtils.isNotEmpty(azureParams.getNetworkId()) && StringUtils.isNotEmpty(azureParams.getResourceGroupName())
+                && MapUtils.isEmpty(subnetMetas)) {
+            String message = String.format("If networkId (%s) and resourceGroupName (%s) are specified then subnet ids must be specified as well.",
+                    azureParams.getNetworkId(), azureParams.getResourceGroupName());
+            LOGGER.info(message);
+            resultBuilder.error(message);
+        }
+    }
+
+    private void checkSubnetIdsAreTheSameAsTheSubnetMetasFromProvider(EnvironmentDto environmentDto, NetworkDto networkDto,
+            ValidationResultBuilder resultBuilder) {
+        if (environmentDto == null || networkDto == null) {
+            LOGGER.warn("EnvironmentDto or NetworkDto. Neither them can be null!");
+            resultBuilder.error("Internal validation error");
+            return;
+        }
+        Map<String, CloudSubnet> cloudmetadata = cloudNetworkService.retrieveSubnetMetadata(environmentDto, networkDto);
+        if (networkDto.getSubnetMetas().size() != cloudmetadata.size()) {
+            String message = String.format("Subnets of the environment (%s) are not found in the vpc (%s).",
```

Comment:
do not use the word vpc in case of azure

Reply:
`vnet` would be more suitable in this context?

Reply:
networkid is fine